### ISSUE TITLE
Simplesamlphp 2.0.2 - autoload_real.php on line 41

### ISSUE THREAD

**Issue by rubenszolt** [simplesamlphp/simplesamlphp#1795]
Fresh install based on https://simplesamlphp.org/docs/stable/simplesamlphp-install-repo.html

**Describe the bug**
Warning: require(/opt/SP/simplesamlphp/vendor/composer/../../tests/_autoload_modules.php): Failed to open stream: No such file or directory in /opt/SP/simplesamlphp/vendor/composer/autoload_real.php on line 41

Fatal error: Uncaught Error: Failed opening required '/opt/SP/simplesamlphp/vendor/composer/../../tests/_autoload_modules.php' (include_path='.:/usr/local/lib/php') in /opt/SP/simplesamlphp/vendor/composer/autoload_real.php:41 Stack trace: #0 /opt/SP/simplesamlphp/vendor/composer/autoload_real.php(45): {closure}('12e461cfb3ae7ce...', '/opt/SP/simples...') #1 /opt/SP/simplesamlphp/vendor/autoload.php(25): ComposerAutoloaderInitab3e9b937a038d427c4eebf0536d6ab2::getLoader() #2 /opt/SP/simplesamlphp/src/_autoload.php(14): require_once('/opt/SP/simples...') #3 /opt/SP/simplesamlphp/public/_include.php(6): require_once('/opt/SP/simples...') #4 /opt/SP/simplesamlphp/public/index.php(7): require_once('/opt/SP/simples...') #5 {main} thrown in /opt/SP/simplesamlphp/vendor/composer/autoload_real.php on line 41

**Comment by rubenszolt** [comment]
I use the docker below for building the image. The build is successfull, however when I try to run the image, it yields the error I have originaly commented.

FROM php/php:8.0-apache-bullseye

ENV TZ=Europe/Budapest
ENV LANG C.UTF-8
ENV SIMPLESAMLPHP_VERSION=2.0.2
ENV COMPOSER_ALLOW_SUPERUSER=1
ARG packages="git curl tar gzip unzip nano libldap2-dev libicu-dev"

USER root

RUN mkdir -p /opt/SP; chmod -R 755 /opt/SP/ && \
addgroup --system --gid 1001 sysuser && \
adduser --home /home/sysuser --shell /bin/bash --uid 1001 --system sysuser --disabled-password

RUN mkdir -p /usr/share/zoneinfo/Europe
COPY timezone/Budapest /usr/share/zoneinfo/Europe/
RUN mkdir -p /root/.composer
RUN for f in /usr/local/share/ca-certificates/*.pem; do cp -- "$f" "${f%.pem}.crt"; done
RUN update-ca-certificates

RUN touch /etc/apt/apt.conf.d/99verify-peer.conf && \
echo >>/etc/apt/apt.conf.d/99verify-peer.conf "Acquire { https::Verify-Peer false }" && \
apt-get -y update && \
apt-get -y install $packages && \
apt-get clean && \
rm -rf /var/lib/apt/lists && \
docker-php-ext-configure ldap --with-libdir=lib/x86_64-linux-gnu/ && \
docker-php-ext-install ldap && \
php --ini && \
docker-php-ext-configure intl --with-libdir=lib/x86_64-linux-gnu/ && \
docker-php-ext-install intl && \
php --ini && \
curl -s "https://github.com/simplesamlphp/simplesamlphp/archive/refs/heads/master.zip" -o /tmp/simplesamlphp-master.zip && \
unzip /tmp/simplesamlphp-master.zip -d /tmp && \
mv /tmp/simplesamlphp-master /opt/SP/simplesamlphp && \
cp /opt/SP/simplesamlphp/config/config.php.dist /opt/SP/simplesamlphp/config/config.php && \
cp /opt/SP/simplesamlphp/config/authsources.php.dist /opt/SP/simplesamlphp/config/authsources.php && \
cp /opt/SP/simplesamlphp/metadata/saml20-idp-hosted.php.dist /opt/SP/simplesamlphp/metadata/saml20-idp-hosted.php && \
cp /opt/SP/simplesamlphp/metadata/saml20-idp-remote.php.dist /opt/SP/simplesamlphp/metadata/saml20-idp-remote.php && \
cp /opt/SP/simplesamlphp/metadata/saml20-sp-remote.php.dist /opt/SP/simplesamlphp/metadata/saml20-sp-remote.php && \
echo

WORKDIR /opt/SP/simplesamlphp
COPY defectdojo-auth/composer.phar /opt/SP/simplesamlphp/composer.phar
RUN php /opt/SP/simplesamlphp/composer.phar clear-cache
#RUN php /opt/SP/simplesamlphp/composer.phar install --prefer-dist -vvv
RUN php /opt/SP/simplesamlphp/composer.phar install -vvv
RUN php /opt/SP/simplesamlphp/composer.phar update -vvv

COPY defectdojo-auth/ports.conf /etc/apache2/ports.conf
COPY defectdojo-auth/simplesamlphp-apache2.conf /etc/apache2/sites-available/simplesamlphp.conf

RUN a2dissite 000-default.conf default-ssl.conf && \
a2enmod rewrite && \
a2ensite simplesamlphp.conf

RUN chmod 777 -R /opt/SP/simplesamlphp/
RUN rm -rf /tmp/*

USER sysuser

EXPOSE 8080

**Comment by tvdijen** [comment]
That's not what I was asking.. You don't expect me to debug your docker file, do you?

**Comment by rmdashrfslash** [comment]
Hi - Having this same issue.  The problem lies in the vendor/composer/autoload_static.php file.  Out of the box, the file works fine.  Once I use composer to install simplesamlphp/simplesamlphp-module-oidc:^3.0.0, the autoload_static file grows significantly and a line is added to the files array:

        '12e461cfb3ae7ce5f3445cf5ccd80c31' => __DIR__ . '/../..' . '/tests/_autoload_modules.php',

**Comment by rmdashrfslash** [comment]
I think I figured it out. Run your composer update with --no-dev.

**Comment by tvdijen** [comment]
The `--no-dev` still doesn't reproduce this issue for me.
Maybe it was a mistake to add composer into the release and we should add it as a phar-file instead.

**Comment by rubenszolt** [comment]
If I download the latest https://github.com/simplesamlphp/simplesamlphp/releases/download/v2.0.2/simplesamlphp-2.0.2.tar.gz and I just unzip, it is working 

But I need to integrate the ldap support.
composer.phar update --no-dev -vvv
composer.phar require --prefer-stable simplesamlphp/simplesamlphp-module-ldap
- the issue persist: Warning: require(/opt/SP/simplesamlphp/vendor/composer/../../tests/_autoload_modules.php): Failed to open stream: No such file or directory in /opt/SP/simplesamlphp/vendor/composer/autoload_real.php on line 41

**Comment by tvdijen** [comment]
I literally did what you said, but I still can't reproduce it;

```
wget https://github.com/simplesamlphp/simplesamlphp/releases/download/v2.0.2/simplesamlphp-2.0.2.tar.gz
tar -xf simplesamlphp-2.0.2.tar.gz
cd simplesamlphp-2.0.2/
composer update --no-dev -vvv
composer require --prefer-stable simplesamlphp/simplesamlphp-module-ldap
```

**Comment by rubenszolt** [comment]
Everything I run in docker but the steps are same.

wget https://github.com/simplesamlphp/simplesamlphp/releases/download/v2.0.2/simplesamlphp-2.0.2.tar.gz
tar -xf simplesamlphp-2.0.2.tar.gz
cd simplesamlphp-2.0.2/
composer update --no-dev -vvv
composer require --prefer-stable simplesamlphp/simplesamlphp-module-ldap
composer clear-cache

RUN curl -s "https://github.com/simplesamlphp/simplesamlphp/releases/download/v2.0.2/simplesamlphp-2.0.2.tar.gz" -o /tmp/simplesamlphp.tar.gzp
RUN tar -xzf /tmp/simplesamlphp.tar.gz -C /tmp
RUN mv /tmp/simplesamlphp-* /opt/SP/simplesamlphp
RUN php /opt/SP/simplesamlphp/composer.phar clear-cache
RUN php /opt/SP/simplesamlphp/composer.phar update --no-dev -vvv
RUN php /opt/SP/simplesamlphp/composer.phar require --prefer-stable simplesamlphp/simplesamlphp-module-ldap
RUN php /opt/SP/simplesamlphp/composer.phar clear-cache

**the issue persist.**
Warning: require(/opt/SP/simplesamlphp/vendor/composer/../../tests/_autoload_modules.php): Failed to open stream: No such file or directory in /opt/SP/simplesamlphp/vendor/composer/autoload_real.php on line 41

Fatal error: Uncaught Error: Failed opening required '/opt/SP/simplesamlphp/vendor/composer/../../tests/_autoload_modules.php' (include_path='.:/usr/local/lib/php') in /opt/SP/simplesamlphp/vendor/composer/autoload_real.php:41 Stack trace: #0 /opt/SP/simplesamlphp/vendor/composer/autoload_real.php(45): {closure}('12e461cfb3ae7ce...', '/opt/SP/simples...') #1 /opt/SP/simplesamlphp/vendor/autoload.php(25): ComposerAutoloaderInit03590d0c044a5d25ee9291b3dc5654e6::getLoader() #2 /opt/SP/simplesamlphp/src/_autoload.php(14): require_once('/opt/SP/simples...') #3 /opt/SP/simplesamlphp/public/_include.php(6): require_once('/opt/SP/simples...') #4 /opt/SP/simplesamlphp/public/index.php(7): require_once('/opt/SP/simples...') #5 {main} thrown in /opt/SP/simplesamlphp/vendor/composer/autoload_real.php on line 41

**Comment by tvdijen** [comment]
Ah but wait, you're not telling the whole story!  You get this error when accessing the `index.php` entry-point, not during the execution composer-commands..

**Comment by tvdijen** [comment]
So, the problem is in your require-command. It will re-install all the dev-dependencies that you told composer not to install in the line before.

Change the line to `composer.phar require --prefer-stable simplesamlphp/simplesamlphp-module-ldap --update-no-dev` and your problems will magically disappear ;)

We do have to fix this on our end as well, because I'm sure you will not be the last person who makes a mistake with composer.

**Comment by tvdijen** [comment]
Fixed in 4f221c21c
